### DISCUSSION TITLE
Better default checkpoint saving path

### DISCUSSION THREAD

**Discussion by APassbyDreg** [open-mmlab/mmengine#1339] [Ideas]
Usually, when I develop a model, a config file is modified many times to start different experiments. Currently, the checkpoints are saved to the default work directory by default. The folder looks like this:

```
config_name/
- timestamp 1/
  - [logs]
- timestamp 2/
  - [logs]
- epoch_xx.pth
```

In this case, it's hard to identify the experiment that generates the checkpoint, and later experiments can overwrite the previous checkpoint.

I suppose that saving the checkpoint to the experiment directory by default is a better idea.

**Comment by APassbyDreg** [comment]
Also, I think that the merged config file should be saved per experiment by default.

**Comment by zhouzaida** [comment]
Hi @APassbyDreg , it is recommended to use different config names for different experiments.
